### PR TITLE
feat: added possibility to start module with C++ code enabled

### PR DESCRIPTION
**Pull request by Patys**
# Summary
 - Script ask you do you want to add C++ code to your module (by default set to "No")
- It adds C++ files and additional code to load it to RN module
- you can use code inside cpp folder to develop your module

## Test Plan

1. Run script
2. Answer last question with Yes (Do you want to use C++ code?)
3. Example contains multiply code

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I mentioned this change in `CHANGELOG.md`
- [x] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: templates/cppLibrary/android/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.4.1)
+
+set (CMAKE_VERBOSE_MAKEFILE ON)
+set (CMAKE_CXX_STANDARD 11)
+
+add_library(cpp
```
Comment (gengjiawen) [inline templates/cppLibrary/android/CMakeLists.txt:6]: Maybe choose another name for library. cpp maybe a little confusing.

File: templates/cppLibrary/android/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.4.1)
+
+set (CMAKE_VERBOSE_MAKEFILE ON)
+set (CMAKE_CXX_STANDARD 11)
```
Comment (gengjiawen) [inline templates/cppLibrary/android/CMakeLists.txt:4]: Maybe 14 ? 11 is bit of old.

File: templates/cppLibrary/android/CMakeLists.txt (hunk 1 above)
Comment (Patys) [reply templates/cppLibrary/android/CMakeLists.txt:6]: Actually it's a good idea. I will try to address that in next pr 👍

File: templates/cppLibrary/android/CMakeLists.txt (hunk 2 above)
Comment (Patys) [reply templates/cppLibrary/android/CMakeLists.txt:4]: I had issues with newer versions as it's compiled for ios and android. I will try to investigate it more in future to bump it.

**Pull request merged**